### PR TITLE
StakingEscrow: function to wrap and topUp stake in threshold staking …

### PR DESCRIPTION
**Pull request by vzotova**
…contract without withdrawing to staker's account

### CODE REVIEW

File: contracts/contracts/StakingEscrow.sol [hunk 1]
```diff
@@ -287,6 +319,34 @@ contract StakingEscrow is Upgradeable, IERC900History {
...
+            info.value
+        );
+
+        uint256 wrappedTokenAmount = info.value - remainder;
+        token.approve(address(vendingMachine), wrappedTokenAmount);
+        vendingMachine.wrap(wrappedTokenAmount);
+        tToken.approve(address(tStaking), tTokenAmount);
+        tStaking.topUp(info.stakingProvider, uint96(tTokenAmount));
+        info.value = remainder;
+        emit WrappedAndTopedUp(msg.sender, wrappedTokenAmount);
```
Comment (derekpierre) [inline contracts/contracts/StakingEscrow.sol:347]:
```suggestion
        emit WrappedAndToppedUp(msg.sender, wrappedTokenAmount);
```

File: contracts/contracts/StakingEscrow.sol [hunk 2]
```diff
@@ -287,6 +319,34 @@ contract StakingEscrow is Upgradeable, IERC900History {
...
+
+        (uint256 tTokenAmount, uint256 remainder) = vendingMachine.conversionToT(
+            info.value
+        );
+
+        uint256 wrappedTokenAmount = info.value - remainder;
+        token.approve(address(vendingMachine), wrappedTokenAmount);
+        vendingMachine.wrap(wrappedTokenAmount);
+        tToken.approve(address(tStaking), tTokenAmount);
+        tStaking.topUp(info.stakingProvider, uint96(tTokenAmount));
```
Comment (derekpierre) [inline contracts/contracts/StakingEscrow.sol:345]: 😍

**Review by derekpierre — APPROVED** [review]
🎸 - very minor comment. Looks great!

File: contracts/contracts/StakingEscrow.sol [hunk 3]
```diff
@@ -287,6 +319,34 @@ contract StakingEscrow is Upgradeable, IERC900History {
...
     }
 
+    /**
+    * @notice Wraps all tokens and top up stake in T staking contract
+    */
+    function wrapAndTopUp() external onlyStaker {
+        StakerInfo storage info = stakerInfo[msg.sender];
+        require(info.stakingProvider != address(0), "There is no stake in T staking contract");
+        require(
+            tStaking.stakedNu(info.stakingProvider) == 0,
```
Comment (cygnusv) [inline contracts/contracts/StakingEscrow.sol:329]: Is it possible that `stakedNu()` will be removed at some point?

Comment (vzotova) [reply contracts/contracts/StakingEscrow.sol:329]: yep, it will be in second upgrade for threshold staking contract for removing legacy

**Pull request merged**